nebula_rust/src/graph_client/session.rs

```diff
@@ -4,4 +4,82 @@
  * attached with Common Clause Condition 1.0, found in the LICENSES directory.
  */
 
-mod graph_client {};
+use crate::graph_client::connection::Connection;
+use crate::graph_client::connection_pool::ConnectionPool;
+
+pub struct Session<'a> {
+    session_id: i64,
+    conn: Connection,
+    pool: &'a ConnectionPool,
+    username: String,
+    password: String,
+    // empty means not a named timezone
+    time_zone_name: String,
+    // Offset to utc in seconds
+    offset_secs: i32,
+    // Keep connection if true
+    retry_connect: bool,
+}
+
+impl<'a> Session<'a> {
+    pub fn new(
+        session_id: i64,
+        conn: Connection,
+        pool: &'a ConnectionPool,
+        username: String,
+        password: String,
+        time_zone_name: String,
+        offset_secs: i32,
+        retry_connect: bool,
+    ) -> Self {
+        Session {
+            session_id: session_id,
+            conn: conn,
+            pool: pool,
+            username: username,
+            password: password,
+            time_zone_name: time_zone_name,
+            offset_secs: offset_secs,
+            retry_connect: retry_connect,
+        }
+    }
+
+    /// sign out the session
+    #[inline]
+    pub async fn signout(&self) -> std::result::Result<(), common::types::ErrorCode> {
```

Comment:
modify the function name to `release`,
and the release function needs to return the connection to pool too.

Reply:
This had done in the drop method.